lib/src/components/files/fileCard/FileCard.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { bool, func } from 'prop-types';
+import Card from 'lib/Card';
+import { createClassNames } from 'helpers/createClassNames';
+
+function FileCard({ thumb, meta, insetMeta, ...cardProps }) {
+  const cardClassNames = createClassNames(
+    'file-card group shadow-small',
+    cardProps
+  );
+
+  const metaContainerClassNames = createClassNames(
+    'border border-neutral-90 border-solid border-b-0 border-l-0 border-r-0',
+    {},
+    {
+      'absolute bottom-0 w-full bg-white -mb-40 group-hover:mb-0 transition-mb duration-200': insetMeta
+    }
+  );
+
+  return (
+    <Card
+      {...cardProps}
+      className={cardClassNames}
+      innerClassNames="relative overflow-hidden rounded"
+    >
+      <Card.Content collapse>{thumb}</Card.Content>
+
+      <Card.Content className={metaContainerClassNames}>{meta}</Card.Content>
```

Comment:
💚 opinion time: i don't really like passing components as props. is there a benefit of doing that here?

Reply:
It might be easier to understand why you don't like passing components as props 😕 

Seeing as in reality there's little difference in passing a component vs a function I don't quite understand your opinion. Components are sugared implementations of `Object.create` so what's the issue?

Reply:
Benefits are;

- You can communicate component UI requirements better.

- Components are functions so you have infinite scope.

- `children` is good but provides no guidance for intent making GCDS language almost impossible to declare in code alone. You're reliant on documentation doing all the hard lifting and that doesn't work.

- Using `children` alone ends up with the dev building consuming app components to provide better standards throughout their app. This ends up either following the pattern here (passing functions) or overpopulation of props which we know doesn't work (e.g. conversations).